packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -2039,6 +2046,854 @@ contract OPContractsManagerInteropMigrator is OPContractsManagerBase {
+            if (res.length != 32) {
+                revert OPContractsManagerV2_ProxyLoadBadReturn();
+            }
+
+            // Will catch if the data is not an abi-encoded address.
+            address result = abi.decode(res, (address));
+
+            // If the address is zero and we must load, revert.
+            if (result == address(0) && _mustLoad) {
+                revert OPContractsManagerV2_ProxyMustLoad();
+            }
+
+            // Return the result.
+            return result;
+        } else {
+            // Handling the failure case well is important to making sure that this function is
+            // safe. This function is great because it cuts down on a large amount of duplicated
+            // code, but we need to make sure that it doesn't go off and deploy contracts when we
+            // don't want it to. This can be potentially unsafe depending on which contract gets a
+            // new proxy deployed.
+
+            // Checking that we "must load" is the first and most important line of defense. The
+            // upgrade path passes in "true" here for contracts by default. This prevents us from
+            // accidentally creating new proxies during an upgrade. Certain upgrades may need to
+            // deploy new proxies for contracts that don't already exist, but we require that all
+            // such cases come with a clear TODO that removes the allowance as soon as possible.
+            if (_mustLoad) {
+                revert OPContractsManagerV2_ProxyMustLoad();
+            }
+
+            // All of the source functions are meant to be simple functions that return a single
+            // address. These functions should not be reverting unless the function doesn't exist
+            // or we ran out of gas (we'll handle that later). It's acceptable for the function not
+            // to exist in certain upgrade paths, but if we get error data back then we hit some
+            // revert statement that we really should be hitting. Only permitted error is
+            // "Proxy: implementation not initialized" from the Proxy contract. We use a hard-coded
+            // bytes32 to reduce code size.
+            if (res.length > 0 && keccak256(res) != 0x1bb64c673de4443083b87fa0508da7efbda546c3bbdd60add6117a438753ffa4)
+            {
+                revert OPContractsManagerV2_ProxyLoadBadError();
+            }
+
+            // Now we check that there's at least 100k gas left. We need this because only 63/64 of
+            // all available gas gets passed into the call above. In theory, it could be possible
+            // for the above call to have reverted without error data if it ran out of gas.
+            // Requiring 100k gas left means that the above call would have needed to use
+            // 63 * 100k = 6.3m gas, which is much larger than we'd ever reasonably expect a source
+            // function to use.
+            if (gasleft() < 100_000) {
```

Comment:
Hmm whenever we have to hardcode gas limits it's worth thinking if this is really needed and how we can make this more robust against gas schedule changes?

Is there a possible future where only 99k gas is needed to complete the top level call and now OPCM breaks? We could make this < 5k gas because the above call should never use more than 315k gas. Or, is it safe to delete this because if we provided too little gas such that the above call reverts, we'll only have 1/64th of the gas left and surely that won't be enough to ever finish executing and the whole tx will revert anyway